packages/mobile/src/components/block-messages/BlockMessages.tsx

```diff
@@ -0,0 +1,163 @@
+  title: 'Are you sure?',
+  confirmText1: 'Are you sure you want to block ',
+  confirmText2: ' from sending messages to your inbox?',
+  info: 'This will not affect their ability to view your profile or interact with your content.',
+  blockUser: 'Block User',
+  cancel: 'Cancel'
+}
+
+const useStyles = makeStyles(({ spacing, typography, palette }) => ({
+  drawer: {
+    marginVertical: spacing(6.5),
+    padding: spacing(3.5)
+  },
+  titleContainer: {
+    ...flexRowCentered(),
+    gap: spacing(3.5),
+    marginBottom: spacing(6),
+    alignSelf: 'center'
+  },
+  title: {
+    fontSize: typography.fontSize.xl,
+    fontFamily: typography.fontByWeight.heavy,
+    color: palette.neutralLight2,
+    textTransform: 'uppercase',
+    lineHeight: typography.fontSize.xl * 1.25,
+    textAlign: 'center'
+  },
+  confirm: {
+    fontSize: typography.fontSize.large,
+    lineHeight: typography.fontSize.large * 1.5,
+    color: palette.neutral
+  },
+  infoContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    gap: spacing(4.5),
+    marginVertical: spacing(4),
+    paddingVertical: spacing(2),
+    paddingHorizontal: spacing(4),
+    backgroundColor: palette.neutralLight9,
+    borderWidth: 1,
+    borderColor: palette.neutralLight7,
+    borderRadius: spacing(2)
+  },
+  infoText: {
+    fontSize: typography.fontSize.medium,
+    lineHeight: typography.fontSize.medium * 1.375,
+    marginRight: spacing(12)
+  },
+  infoIcon: {
+    width: spacing(5),
+    height: spacing(5),
+    color: palette.neutral
+  },
+  button: {
+    padding: spacing(4),
+    height: spacing(12)
+  },
+  blockButton: {
+    borderColor: palette.accentRed
+  },
+  blockText: {
+    fontSize: typography.fontSize.large
+  },
+  cancelButton: {
+    marginTop: spacing(5.5)
+  }
+}))
+
+export const BlockMessagesDrawer = () => {
+  const styles = useStyles()
+  const neutralLight2 = useColor('neutralLight2')
+  const neutral = useColor('neutral')
+  const dispatch = useDispatch()
+  const { userId } = useSelector((state: AppState) =>
+    getData<'BlockMessages'>(state)
+  )
+  const user = useSelector((state) => getUser(state, { id: userId }))
+
+  const handleConfirmPress = () => {
+    dispatch(
+      setVisibility({
+        drawer: 'BlockMessages',
+        visible: false
+      })
+    )
+  }
+
+  const handleCancelPress = () => {
+    dispatch(
+      setVisibility({
+        drawer: 'BlockMessages',
+        visible: false
+      })
+    )
+  }
+
+  return (
+    <NativeDrawer drawerName={BLOCK_MESSAGES_MODAL_NAME}>
+      <View style={styles.drawer}>
+        <View style={styles.titleContainer}>
+          <IconBlockMessages fill={neutralLight2} />
+          <Text style={styles.title}>{messages.title}</Text>
+        </View>
+        <Text style={styles.confirm}>
+          {messages.confirmText1}
+          {user?.name}
+          {messages.confirmText2}
+        </Text>
+        <View style={styles.infoContainer}>
```

Comment:
super nit: I've been seeing this component (the info icon + some text in a box) in some PremiumContent stuff (at least on web) - might be useful to ask if this is going to become a reusable thing, and break it out to its own component if so